src/Qowaiv.Validation.Abstractions/ResultActExtensions.cs

```diff
@@ -0,0 +1,148 @@
+﻿using System;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Qowaiv.Validation.Abstractions
+{
+    /// <summary>Extensions on asynchronous <see cref="Result{TModel}"/>.</summary>
+    public static class ResultActExtensions
+    {
+        /// <summary>Invokes the action when <see cref="Result{TModel}"/> is valid.</summary>
+        /// <param name="promise">
+        /// The promise of the result.
+        /// </param>
+        /// <param name="action">
+        /// The action to invoke.
+        /// </param>
+        /// <typeparam name="TModel">
+        /// The type of input result.
+        /// </typeparam>
+        /// <typeparam name="TOut">
+        /// The type of the new result value.
+        /// </typeparam>
+        /// <returns>
+        /// A result with the merged messages.
+        /// </returns>
+        public static async Task<Result<TOut>> ActAsync<TModel, TOut>(this Task<Result<TModel>> promise, Func<TModel, Result<TOut>> action)
+        {
+            _ = Guard.NotNull(promise, nameof(promise));
```

Comment:
Why are you setting this guard to a variable?

Reply:
To resolve CS4014: Because this call is not awaited, execution of the current method continues before the call is completed. Consider applying the 'await' operator to the result of the call.